src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -142,21 +131,25 @@ public abstract class PeerConnector : IPeerConnector
         }
 
         /// <summary>Constructor used for unit testing.</summary>
-        protected PeerConnector(NodeSettings nodeSettings, IPeerAddressManager peerAddressManager)
+        protected PeerConnector(ILoggerFactory loggerFactory, NodeSettings nodeSettings, IPeerAddressManager peerAddressManager)
         {
             Guard.NotNull(nodeSettings, nameof(nodeSettings));
             Guard.NotNull(peerAddressManager, nameof(peerAddressManager));
 
+            this.connectedPeers = new NetworkPeerCollection();
```

Comment:
I guess we can keep this as null as its set in `Initialize` does it make sense to put an instance here?

Reply:
@dangershony I've removed this constructor in PR #891 👍